keyserver/listener.go

```diff
@@ -0,0 +1,106 @@
+package keyserver
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net"
+	"time"
+
+	. "github.com/coniks-sys/coniks-go/protocol"
+)
+
+func (server *ConiksServer) listenForRequests(ln net.Listener, handler func(msg []byte) ([]byte, error)) {
+	defer ln.Close()
+
+	for {
+		select {
+		case <-server.stop:
+			server.waitCloseConn.Wait()
+			return
+		default:
+		}
+
+		conn, err := ln.Accept()
```

Comment:
A work around for a TCP connection is 
```
    go func() {
		<-server.stop
		ln.(*net.TCPListener).SetDeadline(time.Now())
	}()
```
but we're using TLS, so ... I'm also thinking about it.